<launch>
    <!-- SIMULATOR FOR ROBOT'S HARDWARE -->
    <include file="$(find hardware)/launch/hardware_simul.launch" />
    <!-- ROBOT'S DESCRIPTION -->
    <param name="robot_description" command="cat '$(find mobile_base)/assets/urdf/minibot.urdf'" />

    <group ns="mobile_base">
        <rosparam command="load" file="$(find mobile_base)/assets/params/movement_params.yaml" />

        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
            <remap from="joint_states" to="/joint_states" />
        </node>

        <node name="mobile_base_node" pkg="mobile_base" type="mobile_base_node" output="screen" />
        <node name="simple_move_node" pkg="mobile_base" type="simple_move_node" output="screen" />
    </group>

    <!-- Node to plot simulated movement's data -->
    <!-- <node name="plotjuggler" pkg="plotjuggler" type="plotjuggler" args="layout '$(find mobile_base)/assets/visualization/config_plot_profiles.xml'" /> -->
</launch>